src/Mvc/Mvc.TagHelpers/src/ComponentTagHelper.cs

```diff
@@ -106,7 +113,17 @@ public async override Task ProcessAsync(TagHelperContext context, TagHelperOutpu
 
             // Reset the TagName. We don't want `component` to render.
             output.TagName = null;
-            output.Content.SetHtmlContent(result);
+
+            if (string.IsNullOrEmpty(OutputName))
+            {
+                output.Content.SetHtmlContent(result);
+            }
+            else
+            {
+                var prerenderCache = PrerenderingHelpers.GetOrCreatePrerenderCache(ViewContext);
```

Comment:
I wonder if we can simply use `IRazorPage.SectionWriters` for this. It takes care of things like throwing an error if a section contents were defined, but not consumed. That way you don't have to define another tag helper, you just use an existing idiom in the Layout

Reply:
Although I don't know if Razor allows you to reference a section that appears later in the same page. Might be something to explore.